### ISSUE TITLE
TPE tuner failed to load nested search space json

### ISSUE THREAD

**Issue by tjlin-github** [microsoft/nni#4810]
**Describe the issue**:

**Environment**:
- NNI version: 2.7
- Training service (local|remote|pai|aml|etc): local
- Client OS: ubuntu 18
- Server OS (for remote mode only):
- Python version:python3.7
- PyTorch/TensorFlow version:
- Is conda/virtualenv/venv used?:
- Is running in Docker?: yes

**Configuration**:
 - Experiment config (remember to remove secrets!):
 - Search space:
```json
{
    "C": { "_type": "choice", "_value": [0.01, 0.1, 1, 10, 100] },
    "kernel": { "_type": "choice", "_value": [
        { "_name": "linear" },
        { "_name": "poly",
            "degree": { "_type": "randint", "_value": [1, 11] },
            "gamma": { "_type": "choice", "_value": ["scale", "auto"] }
        },
        { "_name": "rbf", 
            "gamma": { "_type": "choice", "_value": ["scale", "auto"] }
        },
        { "_name": "sigmoid",
            "gamma": { "_type": "choice", "_value": ["scale", "auto"] }
        }
    ]},
    "probability": {"_type":"choice", "_value": [true]}
}
```

**Log message**:
 - nnimanager.log:
 - dispatcher.log:
 - nnictl stdout and stderr:
 - trail log:
```
[2022-04-26 15:16:58] PRINT {'C': 0.1, 'kernel': OrderedDict([('_name', 'sigmoid'), ('gamma', 'auto')]), 'probability': True}
[2022-04-26 15:16:58] ERROR :
  File "/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py", line 255, in fit
    fit(X, y, sample_weight, solver_type, kernel, random_seed=seed)
  File "/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py", line 333, in _dense_fit
    random_seed=random_seed,
  File "sklearn/svm/_libsvm.pyx", line 173, in sklearn.svm._libsvm.fit
ValueError: OrderedDict([('_name', 'sigmoid'), ('gamma', 'auto')]) is not in list
```
 
**How to reproduce it?**: 
Use this search space to train the sklearn.svm.SVC model
It is executable in version 2.5, but this problem occurs in version 2.7

**Comment by liuzhe-lz** [comment]
Can you share the code about how you use the hyperparameters?

**Comment by tjlin-github** [comment]
I convert the parameters given by nni into the model parameters.
-------------------------------
def convert_params(input):
    params = {}
    for key, value in input.items():
        if type(value) == dict:
            params[key] = value.pop('_name')
            params = { **params, **convert_params(value) }
        else: params[key] = value
    return params

params = convert_params(nni.get_next_parameter())
model = sklearn.svm.SVC(**params)

**Comment by Lijiaoa** [comment]
This issue had been fixed in v2.8
